x/evm/keeper/bank.go

```diff
@@ -0,0 +1,123 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v19/x/evm/types"
+)
+
+var _ types.BankWrapper = BankWrapper{}
+
+// BankWrapper is a wrapper around the Cosmos SDK bank keeper
+// that is used to manage an evm denom with 6 or 18 decimals.
+// The wrapper makes the corresponding conversions to achieve:
+//   - With the EVM, the wrapper works always with 18 decimals.
+//   - With the Cosmos bank module, the wrapper works always
+//     with the bank module decimals (either 6 or 18).
+type BankWrapper struct {
+	bk       types.BankKeeper
```

Comment:
The embedded approach is cleaner

```suggestion
	types.BankKeeper
```